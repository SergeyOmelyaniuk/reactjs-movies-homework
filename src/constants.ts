import { Language } from './types';

export const languages: Language[] = [
	{ title: 'en', value: 'en-US' },
	{ title: 'ru', value: 'ru-RU' },
];

export const translate: any = {
	en: {
		title: 'title',
		placeholder: 'Movies, person, movie theaters',
		categories: [
			{ id: 1, value: 'popular', title: 'Popular' },
			{ id: 2, value: 'top_rated', title: 'Top rated' },
			{ id: 3, value: 'upcoming', title: 'Upcoming' },
		],
	},
	ru: {
		title: 'заголовок',
		placeholder: 'Фильмы, люди, кинотеатры',
		categories: [
			{ id: 1, value: 'popular', title: 'Популярные' },
			{ id: 2, value: 'top_rated', title: 'Топ рейтинга' },
			{ id: 3, value: 'upcoming', title: 'Ближайшие' },
		],
	},
};
