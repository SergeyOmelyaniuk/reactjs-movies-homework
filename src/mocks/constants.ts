import { Category, Genre, Language } from '../types';

export const categories: Category[] = [
	{ id: 1, value: 'popular', title: 'Popular' },
	{ id: 2, value: 'top_rated', title: 'Top rated' },
	{ id: 3, value: 'upcoming', title: 'Upcoming' },
];

export const languages: Language[] = [
	{ title: 'en', value: 'en-US' },
	{ title: 'ru', value: 'ru-RU' },
];

export const genres: Genre[] = [
	{ id: 1, name: 'Genre 1' },
	{ id: 2, name: 'Genre 2' },
	{ id: 3, name: 'Genre 3' },
	{ id: 4, name: 'Genre 4' },
	{ id: 5, name: 'Genre 5' },
];
