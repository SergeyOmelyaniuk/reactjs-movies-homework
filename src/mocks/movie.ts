import { MovieDetails } from './../types';

export const movieDescription: MovieDetails = {
	poster_path: 'film_1.jpg',
	genres: [{ id: 1, name: 'first genre' }],
	title: 'Красотка на взводе',
	release_date: '2021-05-26',
	vote_average: 68,
	revenue: 120000,
	runtime: 120,
	overview:
		'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
};
