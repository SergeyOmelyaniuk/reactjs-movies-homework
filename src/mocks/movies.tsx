import { Movie } from '../types';

export const movies: Movie[] = [
	{
		id: 1,
		title: 'Красотка на взводе',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_1.jpg',
		vote_average: 68,
		popularity: 0.5,
		revenue: 120000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [
			{
				name: 'Kate Beckinsale',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_1.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Lindy',
			},
			{
				name: 'Stanley Tucci',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_2.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Dr. Ivan Munchin',
			},
			{
				name: 'Bobby Cannavale',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_3.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Detective Vicars',
			},
			{
				name: 'Jai Courtney',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_4.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Justin',
			},
			{
				name: 'Laverne Cox',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_5.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Detective Nevin',
			},
			{
				name: 'David Bradley',
				birthday: '1959-04-15',
				placeBirth: 'Brest, Belarus',
				biography: 'Text',
				photo: 'photo_6.jpg',
				photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
				character: 'Gareth Fizel',
			},
		],
	},
	{
		id: 2,
		title: 'Сокровище Счастливчика',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_2.jpg',
		vote_average: 72,
		popularity: 0.5,
		revenue: 230000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 3,
		title: 'Сталинград',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_3.jpg',
		vote_average: 76,
		popularity: 0.5,
		revenue: 240000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 4,
		title: 'Люди в черном: Интернэшнл',
		release_date: '2021-05-26',
		genre: ['Боевик', 'Комедия'],
		img: 'film_4.jpg',
		vote_average: 69,
		popularity: 0.5,
		revenue: 136000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
	{
		id: 5,
		title: 'Значит, война',
		release_date: '2021-05-26',
		genre: ['Боевик'],
		img: 'film_5.jpg',
		vote_average: 96,
		popularity: 0.5,
		revenue: 226000,
		duration: '2:14',
		overview:
			'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
		images: [
			'img_1.jpg',
			'img_2.jpg',
			'img_3.jpg',
			'img_4.jpg',
			'img_5.jpg',
			'img_6.jpg',
			'img_7.jpg',
			'img_8.jpg',
		],
		actors: [],
	},
];
