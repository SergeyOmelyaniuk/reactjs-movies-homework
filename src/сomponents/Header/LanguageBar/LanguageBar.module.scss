@import '../../../scss/variables.scss';

.languageBar {
	position: relative;
	background-color: $colorBackgroundHeader;
	@media (max-width: 800px) {
		order: -1;
		margin-bottom: 20px;
	}
}

.menu {
	font-size: 20px;
	display: flex;
	justify-content: center;
	align-items: center;
	align-content: center;
	border: 1px solid $colorWhite;
	border-radius: 3px;
	padding: 7px 10px;
	transition: linear 0.1s;
	color: $colorWhite;
	font-weight: 600;
	text-transform: uppercase;
	cursor: pointer;
	&:hover {
		color: black;
		background-color: $colorWhite;
		border: 1px solid $colorBackgroundHeader;
		border-radius: 3px;
	}
	@media (max-width: 1600px) {
		font-size: 16px;
		padding: 6px 8px;
	}
}

.popup {
	display: flex;
	position: absolute;
	background-color: $colorWhite;
	color: $colorBlack;
	font-size: 20px;
	width: 80px;
	height: 100px;
	right: -16px;
	top: 64px;
	border-radius: 0px 0px 5px 5px;
	text-transform: uppercase;
	flex-direction: column;
	justify-content: center;
	text-align: center;
	z-index: 1;
	&::after {
		content: '';
		position: absolute;
		right: 32px;
		top: -36px;
		display: block;
		border: 8px solid transparent;
		border-bottom: 10px solid $colorWhite;
		transform: translateY(100%);
		@media (max-width: 1600px) {
			right: 26px;
			top: -30px;
			border-bottom: 8px solid $colorWhite;
		}
	}
	@media (max-width: 1600px) {
		font-size: 16px;
		width: 66px;
		height: 83px;
		right: -13px;
		top: 53px;
	}
}

.item {
	cursor: pointer;
	width: 100%;
	&:first-child {
		margin-bottom: 20px;
		@media (max-width: 1600px) {
			margin-bottom: 16px;
		}
	}
	&:hover {
		font-weight: 700;
	}
}
