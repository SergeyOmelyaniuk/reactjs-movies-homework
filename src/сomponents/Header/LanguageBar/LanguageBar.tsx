import React, { useState } from 'react';
import { Language } from '../../../types';
import styles from './LanguageBar.module.scss';

interface LanguageBarProps {
	languages: Language[];
	currentlanguage: Language;
	setCurrentlanguage: (value: Language) => void;
}

function LanguageBar(props: LanguageBarProps) {
	const [isPopupVisible, togglePopup] = useState(false);

	const handler = (lang: Language) => {
		if (props.currentlanguage.title !== lang.title) {
			props.setCurrentlanguage(lang);
			togglePopup(false);
		}
	};

	return (
		<div className={styles.languageBar}>
			<div onClick={() => togglePopup(!isPopupVisible)} className={styles.menu}>
				{props.currentlanguage.title}
			</div>
			{isPopupVisible && (
				<ul className={styles.popup} role='menubar'>
					{props.languages.map((lang) => (
						<li
							onClick={() => handler(lang)}
							key={lang.title}
							className={styles.item}
							role='menuitem'
						>
							{lang.title}
						</li>
					))}
				</ul>
			)}
		</div>
	);
}

export default LanguageBar;
