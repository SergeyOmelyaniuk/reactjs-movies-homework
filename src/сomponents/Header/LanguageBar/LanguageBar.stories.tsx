import React from 'react';

import { ComponentStory, ComponentMeta } from '@storybook/react';
import LanguageBar from './LanguageBar';
import { languages } from '../../../mocks/constants';

export default {
	title: 'Components/LanguageBar',
	component: LanguageBar,
	argTypes: {},
} as ComponentMeta<typeof LanguageBar>;

const Template: ComponentStory<typeof LanguageBar> = (args) => (
	<div
		style={{
			width: '100px',
			height: '200px',
			backgroundColor: '#202020',
			display: 'flex',
			justifyContent: 'center',
		}}
	>
		<LanguageBar {...args} />
	</div>
);

export const Default = Template.bind({});
Default.args = {
	languages: languages,
};
