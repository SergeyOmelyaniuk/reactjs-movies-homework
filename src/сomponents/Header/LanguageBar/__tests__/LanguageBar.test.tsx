import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import LanguageBar from '..';
import { languages } from '../../../../mocks/constants';

describe('LanguageBar', () => {
	test('should render', () => {
		const { container } = render(<LanguageBar languages={languages} />);

		expect(container).toMatchSnapshot();
	});

	test('should render popup on click event', () => {
		const { getByText, queryByRole } = render(
			<LanguageBar languages={languages} />
		);

		expect(queryByRole('menubar')).not.toBeInTheDocument();

		const menuElement = getByText(languages[0].title);
		fireEvent.click(menuElement);

		expect(queryByRole('menubar')).toBeInTheDocument();
	});

	test('should not close popup on click on the same value', () => {
		const { getByText, queryAllByRole, queryByRole } = render(
			<LanguageBar languages={languages} />
		);

		const menuElement = getByText(languages[0].title);
		fireEvent.click(menuElement);

		const items = queryAllByRole('menuitem');
		fireEvent.click(items[0]);

		expect(queryByRole('menubar')).toBeInTheDocument();
	});

	test('should close popup on click on new value', () => {
		const { getByText, queryAllByRole, queryByRole } = render(
			<LanguageBar languages={languages} />
		);

		const menuElement = getByText(languages[0].title);
		fireEvent.click(menuElement);

		const items = queryAllByRole('menuitem');
		fireEvent.click(items[1]);

		expect(queryByRole('menubar')).not.toBeInTheDocument();
		expect(getByText(languages[1].title)).toBeInTheDocument();
	});
});
