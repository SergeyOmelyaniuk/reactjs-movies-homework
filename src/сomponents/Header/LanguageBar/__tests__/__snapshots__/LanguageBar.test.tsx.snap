// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageBar should render 1`] = `
<div>
  <div
    class="languageBar"
  >
    <div
      class="menu"
    >
      en
    </div>
  </div>
</div>
`;
